src/usage.rs

```diff
@@ -0,0 +1,79 @@
+use request::ToParams;
+
+#[derive(Debug)]
+pub struct Usage {
+    metrics: Vec<(String, String)>,
+    usage_params: Vec<(String, String)>,
```

Comment:
There's probably a better way of doing this, but not sure how.

`to_params` is defined in the `ToParams` trait as `fn to_params(&self) -> Vec<(&str, &str)>`. I'm not sure how to return a &str without pre-calculating the parameters.